docs/source-2.0/java/client/generating-clients.rst

```diff
@@ -0,0 +1,202 @@
+==================
+Generating Clients
+==================
+
+The Smithy Java :ref:`build plugin <plugins>`, ``java-client-codegen``, generates Java clients from Smithy models,
+and can be executed with both `Gradle <https://gradle.org/>`_ (recommended) or the :ref:`Smithy CLI <smithy-cli>`.
+
+.. admonition:: Important
+    :class: note
+
+    The Smithy CLI is a prerequisite for this guide when using the ``smithy init``` commands.
+    See the :doc:`Smithy CLI installation guide <../../guides/smithy-cli/cli_installation>`
+    if you do not already have the CLI installed.
+
+-----------------------------------
+Initial setup: Gradle (recommended)
+-----------------------------------
+
+To generated a Java client for a service, first create a new Smithy Gradle project.
+
+.. tip::
+
+    Use the ``smithy init``` CLI command to create a new Smithy project.
+    The command ``smithy init -t quickstart-gradle`` will create a new
+    basic Smithy Gradle project.
+
+Apply the `smithy-base`_ plugin to your Gradle build script to build your Smithy model
+and execute build plugins.
```

Comment:
```suggestion
Apply the `smithy-base`_ plugin to your Gradle build script to build your Smithy model
and execute build plugins:
```

Reply:
Check all of your statements before a code-block for a colon instead of a period.